ocaml/xapi/xapi_session.ml

```diff
@@ -257,6 +257,11 @@ let xapi_internal_originator = "xapi"
 
 let serialize_auth = Mutex.create ()
 
+let assert_username uname = 
+  match uname with
+  | "" -> raise (Api_errors.(Server_error (session_authentication_failed, ["Username is required to login"]))) 
```

Comment:
```suggestion
  | "" -> raise (Api_errors.(Server_error (session_authorization_failed, ["", "Username is required to login"]))) 
```

'session_authentication_failed' is not declared to take any parameters in 'datamodel_errors.ml', but 'session_authorization_failed' does (it takes a username and an error message as parameter).

Reply:
But this is an authentication error, not an authorization error...

Reply:
I think we need to update the error definition in datamodel_errors.ml to take two arguments: the username and a message, as actually most of current cases where this error is raise (several in the same file), these arguments are actually included.

Reply:
In my opinion, an empty string should not be a valid username, and it will be beneficial to prevent empty string from being passed to authentication server and response with a friendly error message. 
I agree with @robhoes , updating error definition in datamode_errors.ml is good solution.

Reply:
@lindig Yes, you are right, empty username does not always fail, like certificate authentication.

The background of this PR is to short circuit the process when try to authenticate to external auth (which is AD domain), which flood AD.  And empty username absolutely have no meaning for Domain Controller.

Reply:
Hi @LunfanZhang Can you please resolve the build error firstly?